bundles/org.openhab.binding.qbus/src/main/java/org/openhab/binding/qbus/internal/handler/QbusDimmerHandler.java

```diff
@@ -0,0 +1,295 @@
+     */
+    public void setSN() {
+        QbusBridgeHandler QBridgeHandler = getBridgeHandler("Dimmer", dimmerId);
+        if (QBridgeHandler == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.CONFIGURATION_ERROR,
+                    "No communication with Qbus Bridge!");
+            return;
+        }
+        this.sn = QBridgeHandler.getSn();
+        ;
+    }
+
+    /**
+     * Handle the status update from the thing
+     */
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        QbusCommunication QComm = getCommunication("Dimmer", dimmerId);
+
+        if (QComm == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Bridge communication not initialized when trying to execute command for dimmer " + dimmerId);
+            return;
+        }
+
+        Map<Integer, QbusDimmer> dimmerComm = QComm.getDimmer();
+        if (dimmerComm != null) {
+            QbusDimmer QDimmer = dimmerComm.get(dimmerId);
+
+            if (QDimmer == null) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                        "Bridge communication not initialized when trying to execute command for dimmer " + dimmerId);
+                return;
+            } else {
+                scheduler.submit(() -> {
+                    if (!QComm.communicationActive()) {
+                        restartCommunication(QComm, "Dimmer", dimmerId);
+                    }
+
+                    if (QComm.communicationActive()) {
+                        if (command == REFRESH) {
+                            handleStateUpdate(QDimmer);
+                            return;
+                        }
+
+                        switch (channelUID.getId()) {
+                            case CHANNEL_SWITCH:
+                                handleSwitchCommand(QDimmer, command);
+                                break;
+
+                            case CHANNEL_BRIGHTNESS:
+                                handleBrightnessCommand(QDimmer, command);
+                                break;
+                        }
+                    }
+                });
+            }
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.CONFIGURATION_ERROR,
+                    "Error while initializing the thing.");
+        }
+    }
+
+    /**
+     *
+     * @param message
+     */
+    public void thingOffline(String message) {
+        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR, message);
+    }
+
+    /**
+     * Executes the switch command
+     */
+    private void handleSwitchCommand(QbusDimmer QDimmer, Command command) {
+        if (command instanceof OnOffType) {
+            OnOffType s = (OnOffType) command;
+            String snr = getSN();
+
+            if (s == OnOffType.OFF) {
```

Comment:
You could directly check for the type. Same for below.
```suggestion
            String snr = getSN();

            if (command == OnOffType.OFF) {
```